### DISCUSSION TITLE
crash at MaxLFQ stage, in re-searches

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#626] [Q&A]
I ran DIANN a first time and at some point it crashed at the second pass MaxLFQ due to no available memory. 
I then re-ran DIANN with --use-quant, after having made sure there is enough disk space. 
However, it just crashed again:

![image](https://user-images.githubusercontent.com/98888664/220971622-76bb30a6-4a12-4fd2-acd3-d48fe762f8fb.png)

I am quite sure it's not MaxLFQ (because the first pass worked and the second pass MaxLFQ has worked for the identical raw files in another search before). 

Due to the crash, I get no saved log, only what it says in the console, see above. 
How to check what the cause is and how to fix it ?

**Comment by vdemichev** [comment]
Well, out of memory might be the case. Swap file enabled? Same happens with --no-maxlfq? 
If you monitor system RAM consumption, is almost all physical RAM consumed before the crash?
Alternative solution is to process stuff in batches.

Best,
Vadim
